packages/ai/src/core.ts

```diff
@@ -76,35 +116,78 @@ export class EdgeDBAI {
     return response;
   }
 
-  async queryRag(
-    message: string,
-    context: QueryContext = this.context,
-  ): Promise<string> {
-    const response = await this.fetchRag({
-      model: this.options.model,
-      prompt: this.options.prompt,
-      context,
-      query: message,
+  async text(): Promise<string> {
+    const res = await this.fetchRag({
+      model: this.model,
+      prompt: this.prompt,
+      context: this.context,
+      query: this.query,
       stream: false,
     });
 
-    if (!response.headers.get("content-type")?.includes("application/json")) {
-      throw new Error(
-        "expected response to have content-type: application/json",
-      );
+    const data = await res.json();
+    return data.response;
+  }
+
+  async stream(): Promise<any> {
+    const res = await this.fetchRag({
+      model: this.model,
+      prompt: this.prompt,
+      context: this.context,
+      query: this.query,
+      stream: true,
+    });
```

Comment:
Since we always pass `this.model`, etc, I think we can simplify `fetchRag` to just take `stream: boolean`.